darli/modeling/integrators/rk4.py

```diff
@@ -0,0 +1,45 @@
+"""Module for RK4 integrator"""
+from .base import Integrator, StateSpaceBase, ArrayLike, cs
+
+
+class RK4(Integrator):
+    @staticmethod
+    def forward(
+        state_space: StateSpaceBase,
+        x0: ArrayLike,
+        qfrc_u: ArrayLike,
+        h: cs.SX | float,
+    ):
+        nq = state_space.model.nq
+        nv = state_space.model.nv
+
+        def state_dot(x, u):
```

Comment:
This function is integrator agnostic, maybe we can put this in to the state space model?